ledger-core/virtual/integration/deduplication/constructor_test.go

```diff
@@ -139,3 +141,206 @@ func TestConstructor_SamePulse(t *testing.T) {
 		})
 	}
 }
+
+func TestDeduplication_DifferentPulses(t *testing.T) {
+	isolation := contract.ConstructorIsolation()
+
+	tests := []struct {
+		Name       string
+		TestRailID string
+		Skipped    string
+
+		VState           payload.VStateReport
+		VStateSendBefore bool
+		VFindCall        *payload.VFindCallResponse
+		VDelegatedCall   *payload.VDelegatedCallRequest
+	}{
+		{
+			Name:       "status is unknown",
+			TestRailID: "C5006",
+			Skipped:    "https://insolar.atlassian.net/browse/PLAT-386",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+			},
+			VStateSendBefore: false,
+			VFindCall: &payload.VFindCallResponse{
+				Status:     payload.UnknownCall,
+				CallResult: nil,
+			},
+		},
+		{
+			Name:       "status is found and result is empty",
+			TestRailID: "C5007",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+			},
+			VStateSendBefore: false,
+			VFindCall: &payload.VFindCallResponse{
+				Status:     payload.FoundCall,
+				CallResult: nil,
+			},
+		},
+		{
+			Name:       "status is found and result is non-empty",
+			TestRailID: "C5008",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+			},
+			VStateSendBefore: false,
+			VFindCall: &payload.VFindCallResponse{
+				Status:     payload.FoundCall,
+				CallResult: &payload.VCallResult{ReturnArguments: []byte("123")},
+			},
+		},
+		{
+			Name:       "status is empty and pending is known",
+			TestRailID: "C5012",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+				ProvidedContent:     &payload.VStateReport_ProvidedContentBody{},
```

Comment:
I don't understand Empty object with 1 pending (constructor). Why does it have any ProvidedContent?

Reply:
fixed